### PR TITLE
fix(backend): Upgrade mysql to 8.0.26

### PR DESCRIPTION
**Pull request by gkcalat**
**Description of your changes:**
Upgrading MySQL to 8.0-debian. Reasons:
- Current MySQL 5.7 has several critical vulnerabilities:
   - [CVE-2020-35527 ](https://security-tracker.debian.org/tracker/CVE-2020-35527)
  - [CVE-2022-37434 ](https://security-tracker.debian.org/tracker/CVE-2022-37434)
  - [CVE-2021-20223 ](https://security-tracker.debian.org/tracker/CVE-2021-20223)
- KFP already offers support for MySQL 8.0 via managed storage options (e.g. on [GCP](https://github.com/GoogleCloudPlatform/kubeflow-distribution/blob/master/kubeflow/apps/pipelines/kustomization.yaml#L24)). This has been already tested.
- Other Kubeflow components already support MySQL 8.0 (e.g. [katib](https://github.com/kubeflow/katib/blob/7bf39225f7235ee4ba6cf285ecc2c455c6471234/manifests/v1beta1/components/mysql/mysql.yaml#L25)).
- MySQL 8.0 is a default version for many platforms (e.g. [GCP](https://cloud.google.com/sql/docs/mysql/db-versions))

This issue is related to GoogleCloudPlatform/kubeflow-distribution/issues/391.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by zijianjoy** [comment]
Looks like MLMD is not able to connect to DB. I am wondering if we need to update https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/base/metadata/overlays/db/metadata-db-deployment.yaml#L24 as well.

**Comment by zijianjoy** [comment]
Looking closer: It seems that we need to set MLMD DB connection properly in https://github.com/kubeflow/pipelines/blob/0a9cf2218ce4fba4a30390f8aaac6de7b7dad318/test/manifests/dev/kustomization.yaml#L13-L16. 

Log: https://pantheon.corp.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22ml-pipeline-test%22%0Aresource.labels.location%3D%22us-east1-b%22%0Aresource.labels.cluster_name%3D%22e2e-bd7142d-20387%22%0Aresource.labels.namespace_name%3D%22kubeflow%22%0Aresource.labels.pod_name%3D%22metadata-grpc-deployment-f8d68f687-wzkhq%22;timeRange=2022-10-11T22:10:03.296Z%2F2022-10-12T05:10:03.296152Z;cursorTimestamp=2022-10-12T01:29:47.678869250Z?project=ml-pipeline-test&jsmode=O&mods=prod_coliseum

File: manifests/gcp_marketplace/chart/kubeflow-pipelines/templates/mysql.yaml [hunk 1]
```diff
@@ -166,9 +166,20 @@ spec:
...
           # https://dev.mysql.com/doc/refman/5.7/en/server-options.html#option_mysqld_ignore-db-dir
           # Ext4, Btrfs etc. volumes root directories have a lost+found directory that should not be treated as a database.
-          - --ignore-db-dir=lost+found
+          # ignore-db-dir option has been deprecated in mysql v5.7.16.
+          #
+          # In order to upgrade MySQL to v8.0 remove /var/lib/mysql/lost+found folder in mysql-pv-claim 
+          # (mysql-persistent-storage):
+          #
+          # kubectl exec -it <mysql-pod-name> -n kubeflow -- bash
+          # rm -rf /var/lib/mysql/lost+found
```
Comment (chensun) [inline manifests/gcp_marketplace/chart/kubeflow-pipelines/templates/mysql.yaml:175]: Chatted offline, we'll see if we can either avoid this or automate it during deployment upgrade.

/hold

Comment (gkcalat) [reply manifests/gcp_marketplace/chart/kubeflow-pipelines/templates/mysql.yaml:175]: Fixed

**Comment by chensun** [comment]
/lgtm
/approve

Thanks @gkcalat !

**Pull request merged**